daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/Pretty.scala

```diff
@@ -21,10 +21,14 @@ private[lf] object Pretty {
         text(s"Scenario failed due to a fetch of an inactive contract") & prettyContractId(coid) &
           char('(') + (prettyIdentifier(tid)) + text(").") &
           text(s"that becomes effective at $effectiveAt")
-      case Error.ContractNotActive(coid, tid, consumedBy) =>
+      case Error.ContractNotActive(coid, tid, Some(consumedBy)) =>
         text("Scenario failed due to a fetch of a consumed contract") & prettyContractId(coid) &
           char('(') + (prettyIdentifier(tid)) + text(").") /
           text("The contract had been consumed in transaction") & prettyEventId(consumedBy)
+      case Error.ContractNotActive(coid, tid, None) =>
+        text("Scenario failed due to a fetch of a consumed contract") & prettyContractId(coid) &
+          char('(') + (prettyIdentifier(tid)) + text(").") /
+          text("The contract had been consumed in transaction")
```

Comment:
For the case where the consumedBy contract ID is non-existent, we probably want to modify the error reporting text here?

Reply:
left a comment.